src/core/types/status.ts

```diff
@@ -6,36 +6,58 @@
  * Side Public License, v 1.
  */
 
-import type { OpsMetrics } from '../server/metrics';
-
-export interface ServerStatus {
-  id: string;
-  title: string;
-  state: string;
-  message: string;
-  uiColor: string;
-  icon?: string;
-  since?: string;
+import type {
+  CoreStatus as CoreStatusFromServer,
+  ServiceStatus as ServiceStatusFromServer,
+  ServiceStatusLevel as ServiceStatusLevelFromServer,
+  OpsMetrics,
+} from '../server';
+
+/**
+ * We need this type to convert the object `ServiceStatusLevel` to a union of the possible strings.
+ * This is because of the "stringification" that occurs when serving HTTP requests.
+ */
+export type ServiceStatusLevel = ReturnType<ServiceStatusLevelFromServer['toString']>;
```

Comment:
This is the biggest change when linking both types: the HTTP response calls the method `toString()`. So the types are not exactly the same.